In [7]:
import numpy as np
import pandas as pd
from sklearn.metrics import mean_absolute_error, root_mean_squared_error, r2_score

In [3]:
daydf = pd.read_csv("../data/raw/day.csv")

In [10]:
daydf["baseline_persistence"] = daydf["cnt"].shift(1)
daydf["baseline_seasonal"] = daydf["cnt"].shift(7)

split = int(len(daydf) * 0.8) # train 80%, test 20%
train = daydf.iloc[:split]
test = daydf.iloc[split:]

In [8]:
# baseline 1: y chang ngày hôm qua
mae1 = mean_absolute_error(test["cnt"], test["baseline_persistence"])
print(f"Persistence baseline MAE: {mae1:.2f}")
rmse = root_mean_squared_error(test["cnt"], test["baseline_persistence"])
print(f"persistence baseline RMSE: {rmse:.2f}")
r2 = r2_score(test["cnt"], test["baseline_persistence"])
print(f"Persistence baseline R2: {r2:.2f}")

Persistence baseline MAE: 878.39
persistence baseline RMSE: 1282.32
Persistence baseline R2: 0.53


In [13]:
# baseline 2: giống ngày này tuần trước
mae2 = mean_absolute_error(test['cnt'], test['baseline_seasonal'])
print(f"Seasonal baseline MAE: {mae2:.2f}")
rmse2 = root_mean_squared_error(test["cnt"], test["baseline_seasonal"])
print(f"Seasonal baseline RMSE: {rmse2:.2f}")
r22 = r2_score(test["cnt"], test["baseline_seasonal"])
print(f"Seasonal baseline R2: {r22:.2f}")

Seasonal baseline MAE: 1194.73
Seasonal baseline RMSE: 1759.66
Seasonal baseline R2: 0.12


# summaryy
- Baseline Persistence đạt MAE thấp hơn Baseline Seasonal (878 so với 1194), do đó được chọn làm baseline chính thức cho các mô hình học máy ở các bước tiếp theo. (mae = 878.39, rmse = 1282.32, r2 = 0.53)